### ISSUE TITLE
rapid_install failing to run becuase of the table db2cv_mview not existing

### ISSUE THREAD

**Issue by AlexsLaboratory** [tripal/tripal#1218]
### **System Info**

- OS: Ubuntu 21.04
- Tripal Version: 3.6
- Drupal 7.81
- PostgreSQL Version: 13
- PHP Version: 7.3

### **Steps to Reproduce**

1.  Install Drupal by downloading the [`drupal-7.81.tar.gz`](https://ftp.drupal.org/files/projects/drupal-7.81.tar.gz) file and moving it into the apache2 web root directory (`/var/www/html`)
2. Install `Drush` by following this [documentation](https://docs.drush.org/en/7.x/install/).
    - Make sure to install the `6.x.x` version as they deprecated a feature used to install `Tripal` in later versions
3. Install `Tripal` using the rapid install [documentation](https://tripal.readthedocs.io/en/latest/user_guide/install_tripal/rapid_install.html)

### **Issue Description**
When the install script runs the `db2cv_mview` table does not get created for some reason so it then tries to create a relation to it and fails to do so because it does not exist.

### **DB Error Message (short)**

```
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR: relation "db2cv_mview" does not exist LINE 1: ...') as name FROM
chado.db DB INNER JOIN db2cv_mvie... ^: SELECT DB.name as short_name, DB.description, DB.url, DB.urlprefix,
SUM(DBCVM.num_terms) as num_terms, array_to_string(array_agg(DBCVM.cvname), ', ') as name FROM chado.db DB INNER 
JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id WHERE DB.name = :name GROUP BY DB.name, DB.description, 
DB.url, DB.urlprefix ; Array ( [:name] => hydra ) in chado_query() (line 1790 of 
/var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
```

### **Install Error Message (detailed)**

```
WD php: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation   [error]
"db2cv_mview" does not exist
LINE 1: ...') as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT
DB.name as short_name, DB.description, DB.url, DB.urlprefix,       
SUM(DBCVM.num_terms) as num_terms,       
array_to_string(array_agg(DBCVM.cvname), ', ') as name      FROM chado.db DB 
     INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE
DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix  
; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of
/var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
Drush command terminated abnormally due to an unrecoverable error.           [error]
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "db2cv_mview" does not exist
LINE 1: ...&#039;) as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT DB.name as short_name, DB.description, DB.url, DB.urlprefix,        SUM(DBCVM.num_terms) as num_terms,        array_to_string(array_agg(DBCVM.cvname), &#039;, &#039;) as name      FROM chado.db DB       INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix   ; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of /var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).

Installing Chado.
Which version of Chado would you like installed?
count(): Parameter must be an array or an object that implements Countable   [warning]
Table.php:699
count(): Parameter must be an array or an object that implements Countable   [warning]
Table.php:699
count(): Parameter must be an array or an object that implements Countable   [warning]
Table.php:699
count(): Parameter must be an array or an object that implements Countable   [warning]
Table.php:699
 [0]  :  Cancel              
 [1]  :  Install Chado v1.3  
 [2]  :  Install Chado v1.2  
 [3]  :  Install Chado v1.11
1
WD php: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation   [error]
"db2cv_mview" does not exist
LINE 1: ...') as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT
DB.name as short_name, DB.description, DB.url, DB.urlprefix,       
SUM(DBCVM.num_terms) as num_terms,       
array_to_string(array_agg(DBCVM.cvname), ', ') as name      FROM chado.db DB 
     INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE
DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix  
; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of
/var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
Drush command terminated abnormally due to an unrecoverable error.           [error]
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "db2cv_mview" does not exist
LINE 1: ...&#039;) as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT DB.name as short_name, DB.description, DB.url, DB.urlprefix,        SUM(DBCVM.num_terms) as num_terms,        array_to_string(array_agg(DBCVM.cvname), &#039;, &#039;) as name      FROM chado.db DB       INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix   ; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of /var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
WD php: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation   [error]
"db2cv_mview" does not exist
LINE 1: ...') as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT
DB.name as short_name, DB.description, DB.url, DB.urlprefix,       
SUM(DBCVM.num_terms) as num_terms,       
array_to_string(array_agg(DBCVM.cvname), ', ') as name      FROM chado.db DB 
     INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE
DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix  
; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of
/var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
Drush command terminated abnormally due to an unrecoverable error.           [error]
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "db2cv_mview" does not exist
LINE 1: ...&#039;) as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT DB.name as short_name, DB.description, DB.url, DB.urlprefix,        SUM(DBCVM.num_terms) as num_terms,        array_to_string(array_agg(DBCVM.cvname), &#039;, &#039;) as name      FROM chado.db DB       INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix   ; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of /var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).

Now preparing the site by creating content types.
WD php: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation   [error]
"db2cv_mview" does not exist
LINE 1: ...') as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT
DB.name as short_name, DB.description, DB.url, DB.urlprefix,       
SUM(DBCVM.num_terms) as num_terms,       
array_to_string(array_agg(DBCVM.cvname), ', ') as name      FROM chado.db DB 
     INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE
DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix  
; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of
/var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
Drush command terminated abnormally due to an unrecoverable error.           [error]
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "db2cv_mview" does not exist
LINE 1: ...&#039;) as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT DB.name as short_name, DB.description, DB.url, DB.urlprefix,        SUM(DBCVM.num_terms) as num_terms,        array_to_string(array_agg(DBCVM.cvname), &#039;, &#039;) as name      FROM chado.db DB       INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix   ; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of /var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
WD php: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation   [error]
"db2cv_mview" does not exist
LINE 1: ...') as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT
DB.name as short_name, DB.description, DB.url, DB.urlprefix,       
SUM(DBCVM.num_terms) as num_terms,       
array_to_string(array_agg(DBCVM.cvname), ', ') as name      FROM chado.db DB 
     INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE
DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix  
; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of
/var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
Drush command terminated abnormally due to an unrecoverable error.           [error]
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "db2cv_mview" does not exist
LINE 1: ...&#039;) as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT DB.name as short_name, DB.description, DB.url, DB.urlprefix,        SUM(DBCVM.num_terms) as num_terms,        array_to_string(array_agg(DBCVM.cvname), &#039;, &#039;) as name      FROM chado.db DB       INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix   ; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of /var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).

Adding permissions for the administrator to view, edit, create, and delete all the newly created content types.
WD php: PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation   [error]
"db2cv_mview" does not exist
LINE 1: ...') as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT
DB.name as short_name, DB.description, DB.url, DB.urlprefix,       
SUM(DBCVM.num_terms) as num_terms,       
array_to_string(array_agg(DBCVM.cvname), ', ') as name      FROM chado.db DB 
     INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE
DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix  
; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of
/var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).
Drush command terminated abnormally due to an unrecoverable error.           [error]
PDOException: SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "db2cv_mview" does not exist
LINE 1: ...&#039;) as name      FROM chado.db DB       INNER JOIN db2cv_mvie...
                                                             ^:       SELECT DB.name as short_name, DB.description, DB.url, DB.urlprefix,        SUM(DBCVM.num_terms) as num_terms,        array_to_string(array_agg(DBCVM.cvname), &#039;, &#039;) as name      FROM chado.db DB       INNER JOIN {db2cv_mview} DBCVM ON DBCVM.db_id = DB.db_id      WHERE DB.name = :name      GROUP BY DB.name, DB.description, DB.url, DB.urlprefix   ; Array
(
    [:name] => hydra
)
 in chado_query() (line 1790 of /var/www/html/drupal/sites/all/modules/tripal/tripal_chado/api/tripal_chado.query.api.inc).

```

**Note:** Everything works fine when you use the [manual install](https://tripal.readthedocs.io/en/latest/user_guide/install_tripal/manual_install.html) process

**Comment by risharde** [comment]
Thanks @Lowe-Man !
Tagging @spficklin who usually works on this as well

**Comment by bradfordcondon** [comment]
looks like `db2cv_mview` is in `setup/tripal_chado.setup.inc` but not in the shcemas which get loaded on basic install, namely, `tripal_chado/chado_schema`.  The table also gets installed in `tripal_chado_update_7323` via `tripal_chado_add_db2cv_mview_mview()`.

Seems to me like the easy solution is to add the `tripal_chado_add_db2cv_mview_mview()` to the `tripal_chado` module install (as dbupdates dont run on initial install).. 

Submitted a quick pull request but im quite rusty.  Thanks for the report.

**Comment by spficklin** [comment]
@Lowe-Man . I could not duplicate the issue on a fresh rapid install, however we did apply a fix in PR #1225 that should fix the problem  for you.   You will probably have to wait until we do an official release with this fix in it,  But I'm assuming at this point you probably got around the error by doing a manual install. If not let us know. I'm going to close this out but please comment if you're still having issues.
